src/main/java/org/aksw/iguana/cc/worker/impl/SPARQLProtocolWorker.java

```diff
@@ -127,21 +129,30 @@ public SPARQLProtocolWorker(long workerId, ResponseBodyProcessor responseBodyPro
      */
     public static void initHttpClient(int threadCount) {
         connectionManager = PoolingAsyncClientConnectionManagerBuilder.create()
-                .setMaxConnTotal(threadCount)
-                .setMaxConnPerRoute(threadCount)
+                .setMaxConnTotal(threadCount * 1000)
+                .setMaxConnPerRoute(threadCount * 1000)
+                .setPoolConcurrencyPolicy(PoolConcurrencyPolicy.LAX)
+                .setConnPoolPolicy(PoolReusePolicy.FIFO)
+                .setDefaultConnectionConfig(org.apache.hc.client5.http.config.ConnectionConfig.custom()
+                        .setConnectTimeout(Timeout.ofSeconds(5))
+                        .build())
                 .build();
         final var ioReactorConfig = IOReactorConfig.custom()
                 .setTcpNoDelay(true)
-                .setIoThreadCount(threadCount)
+                .setIoThreadCount(threadCount + 1)
+                .setSelectInterval(TimeValue.ofMilliseconds(100))
+                .setSoKeepAlive(true)
                 .build();
         httpClient = HttpAsyncClients.custom()
                 .setConnectionManager(connectionManager)
                 .setIOReactorConfig(ioReactorConfig)
                 .setKeepAliveStrategy(new DefaultConnectionKeepAliveStrategy())
+                .setConnectionManagerShared(false)
                 .setDefaultRequestConfig(RequestConfig.custom()
                         .setContentCompressionEnabled(false)
                         .setHardCancellationEnabled(true)
                         .build())
+                .setRetryStrategy(DefaultHttpRequestRetryStrategy.INSTANCE)
```

Comment:
This means one retry after 1s, see https://hc.apache.org/httpcomponents-client-5.2.x/current/httpclient5/apidocs/org/apache/hc/client5/http/impl/DefaultHttpRequestRetryStrategy.html#DefaultHttpRequestRetryStrategy--

I think there should be no retries. If the server fails to answer the query is failed.